Automattic/wp-calypso
# Add GitHub button skeleton

Closes #https://github.com/Automattic/wp-calypso/issues/87037

## Proposed Changes

This PR adds a basic "Continue with GitHub" button component. Functionality wise:

- It renders a button (only when the feature flag is enabled)
- It accounts for three ways for it to be in a disabled state (Redux, local state & error). 
- It allows the passing of a child, which will be used as a custom button (needed for `/me/security/social-login`)
- It allows for the rendering of an error in a popover. 

It does not: 

- Anything related to the actual login flow. This will be handled in a future PR.
 
<img width="601" alt="image" src="https://github.com/Automattic/wp-calypso/assets/528287/82aa0345-ce61-4cf2-acaa-97d7aeab2c1c">


## Testing Instructions

- Apply this PR
- Using a signed-out tab, visit: http://calypso.localhost:3000/start
- You should be presented with a sign up screen, containing the "Continue with GitHub" button
- Click "Log in" in the top right.
- You should be presented with a login screen, containing the "Continue with GitHub" button

Verify that the button doesn't render in any other environment except for development (for example by visiting the Calypso Live link).

## Pre-merge Checklist
